### DISCUSSION TITLE
How can I debug condition_variable?

### DISCUSSION THREAD

**Discussion by y-ich** [emscripten-core/emscripten#21141] [Q&A]
Hi.

I compiled relatively large program by emscripten 3.1.52.
The wait_for method of some condition_variable in a thread never return even after duration of second argument.
Only pred of third argument is effective.

How can I debug it?

I tried to reproduce it with simple codes, but they work well.

Thanks.

**Comment by sbc100** [comment]
Are you observing that `wait_for` does not return after the specified timeout in general?  Or is this something that is intermittent or only effects a certain `condition_variable`?

Is this a regression?  (i.e. did it work in some previous version of emscripten?)

As for debugging you could try using the Wasm debugger chrome extension (https://developer.chrome.com/blog/wasm-debugging-2020).  If you attempt to do any debugging via print statements you might want to use the `emscripten_out` and `emscripten_err` helpers rather then printf since they log more directly.

Is your program specific to the web or can it also run under node?  If so can you repro the issue under node?  (some things can be easier to do debug there).

**Comment by y-ich** [comment]
@sbc100 san, thank you for your comments.

I am sorry that my issue was totally my misunderstanding. Wait_for of condition_variable works well.
